lib/elixir/lib/task/supervised.ex

```diff
@@ -200,44 +202,51 @@ defmodule Task.Supervised do
        when max == 0
        when next == :done do
     %{monitor_pid: monitor_pid, monitor_ref: monitor_ref,
-      timeout: timeout, on_timeout: on_timeout} = config
+      timeout: timeout, on_timeout: on_timeout, ordered: ordered?} = config
 
     receive do
       # The task at position "position" replied with "value". We put the
       # response in the "waiting" map and do nothing, since we'll only act on
       # this response when the replying task dies (we'll notice in the :down
       # message).
-      {{^monitor_ref, position}, value} ->
-        %{^position => {pid, :running}} = waiting
-        waiting = Map.put(waiting, position, {pid, {:ok, value}})
+      {{^monitor_ref, position}, reply} ->
+        %{^position => {pid, :running, value}} = waiting
```

Comment:
What if instead of `value` we use something like `enum_element` or something like this? Something that suggests this is the original element in the enumerable.